### DISCUSSION TITLE
Белый список хостов

### DISCUSSION THREAD

**Discussion by Frosty726** [bol-van/zapret#618] [Q&A]
Пытался найти, как исключить некоторые хосты. Точнее говоря, мне надо было исключить `tbank.ru`.
Нашёл в `readme.txt` про файл `zapter-hosts-user-exclude.txt`.
Создал его, добавил `tbank.ru`, запустил `get_config.sh`, чтобы создался `zapret-ip-exclude.txt`.
Запускал скрипт, как было - не сработало.
Добавил аргумент `--ipset-exclude=<filename>` и вбил абсолютный путь до `zapret-ip-exclude.txt` - не помогло.
Возможно, я что-то делаю неправильно или дело в `tbank.ru` и надо добавлять ещё какие-то домены?

**Comment by numbereleven-a** [comment]
а как вы проверяете что что для данного хоста идёт дурение после добавления в exclude?
плюс стратегия с 0 фазой вроде не будет работать на листы

**Reply by Frosty726** [reply]
Проверка простая: захожу на сайт - он не доступен.
Все настройки в install_easy.sh стандартные, доп. настройки такие:
```
TPWS_OPT="--hostspell=hoSt --split-http-req=method --split-pos=3 --oob"
TPWS_OPT_SUF=""
```
Я так понимаю, никакой 0 фазы тут нет (если она не задана где-то по умолчанию).

**Reply by numbereleven-a** [reply]
> Проверка простая: захожу на сайт - он не доступен. Все настройки в install_easy.sh стандартные, доп. настройки такие:
> 
> ```
> TPWS_OPT="--hostspell=hoSt --split-http-req=method --split-pos=3 --oob"
> TPWS_OPT_SUF=""
> ```
> 
> Я так понимаю, никакой 0 фазы тут нет (если она не задана где-то по умолчанию).

не заметил что у тебя ip в txt, надо в zapret-hosts-user-exclude.txt добавлять

**Reply by Frosty726** [reply]
В `zapret-hosts-user-exclude.txt` `tbank.ru` добавлен уже. И скрипт `get_config.sh` тоже запускал, чтобы обновить файл `zapret-ip-exclude.txt`.

**Reply by numbereleven-a** [reply]
понял
может попробовать --debug и посмотреть там?

**Comment by bol-van** [comment]
параметры --ipset и --ipset-exclude существуют только для виндовс и андроид. на других системах он противопоказан

**Reply by Frosty726** [reply]
Спасибо, это поправил, т.к. у меня poenwrt. Но и без этого исключающий список не подцеплялся. Возможно, стоит использовать фильтрацию через ipset, чтобы всё заработало? Но в этом варианте, как я понял, надо наоборот создавать список всех ip, к которым применяется Zapret.

**Reply by bol-van** [reply]
zapret-hosts-exclude при любых конфигурациях ресолвится в zapret-ip-exclude и помещается в сет nozapret.
Все правила создаются с учетом этого сета

nft list table inet zapret | grep nozapret
nft list set inet zapret nozapret

и все станет понятно.

Но многие крупные сайты располагаются на CDN и имеют прыгающие IP. Фильтрация по ipset будет нестабильна.
tbank вроде к ним не относится. Со всех днс ресолвится одно и тоже.

И, конечно, как правильно заметили, веб сайт может обращаться к разным доменам. К каким именно видно по F12 в броузере.

**Reply by CalamityDeadshot** [reply]
Почему?

**Reply by bol-van** [reply]
Перенаправление трафика - штука тяжелая. Если можно отсечь фаерволом - надо отсекать

**Reply by agbtar** [reply]
вы сейчас мне конечно расскажите про обрезанные файрволы, но нужен ли козе баян?

пускай нет нормального файрвола на винде и андройд - нет и фиг с ним и не надо. запрет должен фильтровать по SNI - читай доменное имя.  потому что наоборот на одном IP может быть несколько SNI соответственно другая фильтрация ненужный хлам.

тут кто-то наверно скажет как быть с ECH? но сайты использующие ECH весят как правило на CDN от клаудфлер - фильтрация по IP тоже будет работать так себе.

какое еще перенаправление трафика? вы о чем? запрет зпущен на пограничном маршрутизаторе?

**Reply by bol-van** [reply]
Перенаправление - это передача пакетов из ядра в очередь средствами nfqueue или divert
Бывают диапазоны адресов, которые заблокированы иначе. CDN, например.
zapret не ограничивается дурением TLS

**Reply by agbtar** [reply]
зато блокировки ограничиваются. не считая блокировок по IP и смешных блокировок по DNS.
вы забили что-то типа:
`add divert 911 ip from ${lan} to not "table(nofrag)" out not diverted xmit em0`
в таблицу напихали IP-шников, можно скачать лист префиксов рунета и захреначит его в таблицу. но это для больших собачек - пользователю с мыльницей с антенной этого не надо. уж в тем более этого не надо пользователю виндовз или андройд.

т.е. вопрос о нужности бояна остался. и, уважаемый автор, фактически ограничивается -  препятствует выделению други сигнатур. весь дополнительный функционал там чтобы делать именно это. запрет средство обхода DPI, а не замещения функций файровола на кастрированных платформах.
